app/scripts/controllers/edit/buildConfig.js

```diff
@@ -117,6 +118,76 @@ angular.module('openshiftConsole')
       "SerialLatestOnly"
     ];
 
+    $scope.buildHookTypes = [{
+      "id": "script",
+      "label": "Shell Script"
+    }, {
+      "id": "command",
+      "label": "Command"
+    }, {
+      "id": "args",
+      "label": "Arguments to Default Image Entry Point"
+    }, {
+      "id": "scriptArgs",
+      "label": "Shell Script with Arguments"
+    }, {
+      "id": "commandArgs",
+      "label": "Command with Arguments"
+    }];
+
+    $scope.buildHookSelection = {
+      "type": {}
+    };
+
+    var getInitialBuildHookSelection = function() {
+      $scope.view.hasHooks = (_.size($scope.buildConfig.spec.postCommit.args) ||
+                              _.size($scope.buildConfig.spec.postCommit.command) ||
+                              $scope.buildConfig.spec.postCommit.script) ? true : false;
+
+      if (_.size($scope.buildConfig.spec.postCommit.args) && _.size($scope.buildConfig.spec.postCommit.command)) {
+        $scope.buildHookSelection.type.id = "commandArgs";
+      } else if ($scope.buildConfig.spec.postCommit.script && _.size($scope.buildConfig.spec.postCommit.args)) {
+        $scope.buildHookSelection.type.id = "scriptArgs";
+      } else if (_.size($scope.buildConfig.spec.postCommit.args)) {
+        $scope.buildHookSelection.type.id = "args";
+      } else if ($scope.buildConfig.spec.postCommit.script) {
+        $scope.buildHookSelection.type.id = "script";
+      } else {
+        $scope.buildHookSelection.type.id = "command";
+      }
+
+      $scope.buildHookSelection.type.label = _.find($scope.buildHookTypes, {'id': $scope.buildHookSelection.type.id}).label;
```

Comment:
Shouldn't need to set this. You can use `{{$select.selected.label}}` in the `ui-select` markup.